File: AtlasWeb/Core/Modules/Tibbi_Surec/Tetkik_Istem_Modulu/ProceduresRequiredInfoViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using TTObjectClasses;

namespace Core.Modules.Tibbi_Surec.Tetkik_Istem_Modulu
{
    public class ProceduresRequiredInfoViewModel
    {
        public List<RadiologyTest> RadiologyTestList = new List<RadiologyTest>();
        public List<LaboratoryProcedure> LaboratoryTestList = new List<LaboratoryProcedure>();
        public List<ProcedureDefinition> ProcedureObjectList = new List<ProcedureDefinition>();
        public List<PsychologyTest> PsychologyProcedureList = new List<PsychologyTest>();
        public List<PathologyTestProcedure> PathologyProcedureList = new List<PathologyTestProcedure>();
        public List<NuclearMedicineTest> NuclearMedicineTestList = new List<NuclearMedicineTest>();
        public List<ManipulationProcedure> ManipulationProcedureList = new List<ManipulationProcedure>();
        public List<RadiologyTestTypeDefinition> RadiologyTestTypeDefinitions = new List<RadiologyTestTypeDefinition>();
        public string ClinicAnemnesis;
        public Boolean copyDescription;
        public string textDescription;
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Tetkik_Istem_Modulu/ProceduresRequiredInfoViewModel.ts
import { RadiologyTest, LaboratoryProcedure, ProcedureDefinition, PsychologyTest, PathologyTestProcedure, NuclearMedicineTest, ManipulationProcedure, RadiologyTestTypeDefinition } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';
export class ProceduresRequiredInfoViewModel {
    @Type(() => RadiologyTest)
    RadiologyTestList: Array<RadiologyTest> = new Array<RadiologyTest>();
    @Type(() => LaboratoryProcedure)
    LaboratoryTestList: Array<LaboratoryProcedure> = new Array<LaboratoryProcedure>();
    @Type(() => ProcedureDefinition)
    ProcedureObjectList: Array<ProcedureDefinition> = new Array<ProcedureDefinition>();
    @Type(() => PsychologyTest)
    PsychologyProcedureList: Array<PsychologyTest> = new Array<PsychologyTest>();
    @Type(() => PathologyTestProcedure)
    PathologyProcedureList: Array<PathologyTestProcedure> = new Array<PathologyTestProcedure>();
    @Type(() => NuclearMedicineTest)
    NuclearMedicineTestList: Array<NuclearMedicineTest> = new Array<NuclearMedicineTest>();
    @Type(() => ManipulationProcedure)
    ManipulationProcedureList: Array<ManipulationProcedure> = new Array<ManipulationProcedure>();
    @Type(() => RadiologyTestTypeDefinition)
    RadiologyTestTypeDefinitions: Array<RadiologyTestTypeDefinition> = new Array<RadiologyTestTypeDefinition>();
    
    ClinicAnemnesis: string;
    @Type(() => Boolean)
    copyDescription: boolean;
    textDescription: string;
}